### PR TITLE
Merge AUM fee getters

### PR DESCRIPTION
**Pull request by TomAFrench**
Builds on #1890 

This PR merges the getters for the AUM fee percentage and timestamp while packing the two values in storage. For bytecode reasons I've introduced (another!) library for managing this state vs using a struct.

### CODE REVIEW

File: pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol [hunk 1]
```diff
@@ -582,17 +576,18 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, IControlled
...
+    {
+        (aumFeePercentage, lastCollectionTimestamp) = ManagedPoolAumStorageLib.getAumFeeFields(_aumState);
 
-    /**
-     * @notice Returns the timestamp of the last collection of AUM fees.
-     */
-    function getLastAumFeeCollectionTimestamp() external view returns (uint256) {
-        return _lastAumFeeCollectionTimestamp;
+        // If we're in recovery mode then we bypass any fee logic by setting the fee percentage to zero.
+        aumFeePercentage = ManagedPoolStorageLib.getRecoveryModeEnabled(_poolState) ? 0 : aumFeePercentage;
```
Comment (nventuro) [inline pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol:590]:
```suggestion
        // If we're in recovery mode then we bypass any fee logic by setting the fee percentage to zero.
        if (ManagedPoolStorageLib.getRecoveryModeEnabled(_poolState)) {
            aumFeePercentage = 0;
        }
```

File: pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol [hunk 2]
```diff
@@ -667,10 +671,9 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, IControlled
             return 0;
         }
 
-        // As we update `_lastAumFeeCollectionTimestamp` when updating `_managementAumFeePercentage`, we only need to
-        // update `_lastAumFeeCollectionTimestamp` when non-zero AUM fees are paid. This avoids an SSTORE on zero-length
-        // collections.
-        _lastAumFeeCollectionTimestamp = block.timestamp;
+        // As we update the AUM fee collection timestamp when updating the AUM fee percentage, we only need to
+        // update the timestamp when non-zero AUM fees are paid. This avoids an SSTORE on zero-length collections.
+        _updateAumFeeCollectionTimestamp();
```
Comment (nventuro) [inline pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol:676]: `getManagementAumFeeParams()` reads `_aumState`, and here we read it again. Bit unfortunate that we need to do the warm sload, but that's likely better than passing state around, or having duplicate functions.

File: pkg/pool-weighted/test/foundry/ManagedPoolAumStorageLib.t.sol [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+        bytes32 newAumState = ManagedPoolAumStorageLib.setAumFeePercentage(aumState, expectedAumFeePercentage);
+        assertOtherStateUnchanged(aumState, newAumState, _AUM_FEE_PERCENTAGE_OFFSET, _AUM_FEE_PCT_WIDTH);
+
+        (uint256 actualAumFeePercentage, ) = ManagedPoolAumStorageLib.getAumFeeFields(newAumState);
+        assertEq(actualAumFeePercentage, expectedAumFeePercentage);
+    }
+
+    function testSetLastCollectionTimestamp(bytes32 aumState, uint256 expectedLastCollectionTimestamp) public {
+        vm.assume(expectedLastCollectionTimestamp < _MAX_TIMESTAMP);
```
Comment (nventuro) [inline pkg/pool-weighted/test/foundry/ManagedPoolAumStorageLib.t.sol:62]:
```suggestion
        vm.assume(expectedLastCollectionTimestamp <= _MAX_TIMESTAMP);
```

File: pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol (hunk 1 above)
Comment (TomAFrench) [reply pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol]: Huh, I had this as it saved bytecode in #1890. Seems like the opposite here though.

File: pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol (hunk 2 above)
Comment (TomAFrench) [reply pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol:663]: uno_reverse_card.jpg

**Pull request merged**